File: java-lotto/src/main/java/dev/hongsii/model/WinningLotto.java
package dev.hongsii.model;

import java.util.List;
import java.util.stream.Collectors;

public class WinningLotto {

    private Lotto winningLotto;

    public WinningLotto(Lotto winningLotto) {
        this.winningLotto = winningLotto;
    }

    public WinningAnalyzer win(List<Lotto> lottos) {
        return new WinningAnalyzer(
                lottos.stream()
                        .map(this::win)
                        .collect(Collectors.toList())
        );
    }

    public WinningResult win(Lotto lotto) {
        int matchCount = winningLotto.match(lotto);
        boolean matchBonus = winningLotto.matchBonus(lotto);
        return WinningResult.of(new MatchResult(matchCount, matchBonus));
    }

    @Override
    public String toString() {
        return winningLotto.toString();
    }
}

File: kotlin-lotto/src/main/kotlin/lotto/model/WinningLotto.kt
package lotto.model

class WinningLotto(private val winningLotto: Lotto, private val bonusNumber: Number) {

    init {
        require(!winningLotto.contains(bonusNumber)) { "보너스 번호가 중복되었습니다." }
    }

    fun win(lotto: Lotto): WinningResult {
        val matchCount = winningLotto.matchWith(lotto)
        val matchBonusNumber = lotto.contains(bonusNumber)
        return WinningResult.of(MatchResult(matchCount, matchBonusNumber))
    }

    fun getLotto() = winningLotto
    fun getBonusNumber() = bonusNumber
}